pkg/services/services.go

```diff
@@ -194,12 +190,57 @@ func klogVerbosityArg(debug bool) string {
 }
 
 // getKubeletHostnameOverride returns the hostname override arg that should be used
-func getKubeletHostnameOverride(platformType string) string {
+func getKubeletHostnameOverride(platformType string) (string, servicescm.PowershellCmdArg) {
 	platformType = strings.ToUpper(platformType)
+	// define argument with fixed variable name
+	hostnameOverrideArg := "--hostname-override=" + hostnameOverrideVarName
 	switch platformType {
 	case string(config.AWSPlatformType):
-		return "--hostname-override=" + ec2HostnameVar
+		return hostnameOverrideArg, getAWSMetadataHostnamePowershellCmdArg(hostnameOverrideVarName)
+	case string(config.GCPPlatformType):
+		return hostnameOverrideArg, getGCPMetadataHostnamePowershellCmdArg(hostnameOverrideVarName)
 	default:
-		return ""
+		return "", servicescm.PowershellCmdArg{}
 	}
 }
+
+// getAWSMetadataHostnamePowershellCmdArg returns the PowershellCmdArg to resolve the hostname using the instance
+// metadata service in AWS. Uses AWS Tools for PowerShell (https://docs.aws.amazon.com/powershell/latest/userguide/pstools-welcome.html)
+func getAWSMetadataHostnamePowershellCmdArg(variableName string) servicescm.PowershellCmdArg {
+	return servicescm.PowershellCmdArg{
+		Name: variableName,
+		Path: "Get-EC2InstanceMetadata -Category LocalHostname",
+	}
+}
+
+// getGCPMetadataHostnamePowershellCmdArg returns the PowershellCmdArg to resolve the hostname using the instance
+// metadata service in GCP. Uses GoogleCloud Tools for PowerShell (https://cloud.google.com/tools/powershell/docs/quickstart)
+// getGCPMetadataHostname returns the GCP instance hostname from the metadata service.  GCP is an especial case,
+// and the resulting hostname is truncated if the length is longer than 63 characters.
+func getGCPMetadataHostnamePowershellCmdArg(variableName string) servicescm.PowershellCmdArg {
+	return servicescm.PowershellCmdArg{
+		Name: variableName,
+		Path: `
```

Comment:
As an alternative, I can move this PowerShell script into a file under the [pkg/internal](https://github.com/openshift/windows-machine-config-operator/tree/master/pkg/internal) directory, add the file to the payload and here just declare the path. Thoughts?

Reply:
I agree with the proposed alternative. Having the script live in a separate file might be a cleaner approach.

Reply:
+1. That is is the pattern we have been following so far.

Reply:
+1

Reply:
To be clear the request is to include this as a file in the internal directory

Reply:
You could check platform before adding it to the files to be transferred in `windows.go`. We currently transfer `azure-cloud-node-manager.exe` to every instance on every platform though, so this may warrant a team-wide consensus on whether everything in the payload should always be transferred as is.

Reply:
+1

And this should not be the reason to not make this file

Reply:
Makes sense, will squash the commits. 

Thanks you all for the feedback.